eng/CodeAnalysis.ruleset

```diff
@@ -214,4 +214,7 @@
         <Rule Id="SA1650"  Action="None" />          <!-- Element documentation should be spelled correctly -->
         <Rule Id="SA1651"  Action="None" />          <!-- Do not use placeholder elements -->
     </Rules>
+    <Rules AnalyzerId="Microsoft.CodeAnalysis.CSharp.NetAnalyzers" RuleNamespace="Microsoft.CodeAnalysis.CSharp.NetAnalyzers">
+        <Rule Id="CA1825" Action="Error" />          <!-- Avoid unnecessary zero-length array allocations -->
```

Comment:
Curious: is this adding the rule so it is applied to generated libraries?

Reply:
No. That's only for projects that reference this ruleset, which are only those projects in this repo. Besides, with `WarningsAsErrors=true` in our azure-sdk-for-net repo, it's already an error for all those projects unless suppressed explicitly.